script

```diff
@@ -2600,21 +2591,23 @@ only_send_error_notifications="0"
         # initialize variables to find number of vdisks
         numberofvdisks="1"
 
+        local _addnl_ext
+        if [ "$inline_zstd_compress" -eq 1 ]; then
+          _addnl_ext=".zst"
+        fi
+
         # find each vdisk extension and use it to build a regular expression to get a vdisk's trailing number.
         for extension in "${vdisk_extensions[@]}"
         do
-          if [ "$inline_zstd_compress" -eq 1 ]; then
-	      extension+=".zst"
-	    fi
 
           # check to see if regular expression is empty.
           if [[ -z "$vdisknumberextregex" ]]; then
             # build intial regular expression.
             vdisknumberextregex='[0-9]+\.('
-            vdisknumberextregex="${vdisknumberextregex}$extension"
+            vdisknumberextregex="${vdisknumberextregex}$extension$_addnl_ext"
           else
             # add additional extensions to regular expression.
-            vdisknumberextregex="${vdisknumberextregex}|$extension"
+            vdisknumberextregex="${vdisknumberextregex}|$extension$_addnl_ext"
           fi
```

Comment:
Since we are adding the zst extension when the array gets created (around line 775), we don't need any reference to it here.
Also, _addnl_ext can't be local here since it is not inside a function.

Reply:
Not sure I understand the first part?  The zst extension is not being added to any array around line 775 that I can see.

But, I just committed a better version anyway.